src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -3607,6 +3476,97 @@ public override bool HandleCall (MethodBody callingMethodBody, MethodReference c
+						// static Property (Expression, Type, String)
+						//
+						case var fieldOrProperty when ((fieldOrProperty == "Field" || fieldOrProperty == "Property")
+							&& calledMethod.DeclaringType.Namespace == "System.Linq.Expressions"
+							&& calledMethod.DeclaringType.Name == "Expression"
+							&& calledMethod.Parameters.Count == 3
+							&& calledMethod.Parameters [1].ParameterType.FullName == "System.Type"): {
+
+								reflectionContext.AnalyzingPattern ();
+
+								foreach (var value in methodParams [1].UniqueValues ()) {
+									if (value is SystemTypeValue systemTypeValue) {
+										foreach (var stringParam in methodParams [2].UniqueValues ()) {
+											if (stringParam is KnownStringValue stringValue) {
+												bool staticOnly = methodParams [0].Kind == ValueNodeKind.Null;
+												// TODO: Change this as needed after deciding if we are to keep all fields/properties on a type
+												// that is accessed via reflection. For now, let's only keep the field/property that is retrieved.
+												if (calledMethod.Name [0] == 'P')
+													MarkPropertiesFromReflectionCall (ref reflectionContext, systemTypeValue.TypeRepresented, stringValue.Contents, staticOnly);
+												else
+													MarkFieldsFromReflectionCall (ref reflectionContext, systemTypeValue.TypeRepresented, stringValue.Contents, staticOnly);
+											} else {
+												reflectionContext.RecordUnrecognizedPattern ($"Expression call '{calledMethod.FullName}' inside '{callingMethodBody.Method.FullName}' was detected with 3rd argument which cannot be analyzed");
+											}
+										}
+									} else if (value == NullValue.Instance) {
+										reflectionContext.RecordHandledPattern ();
+									} else if (value is MethodParameterValue) {
+										// TODO: Check if parameter is annotated.
+										reflectionContext.RecordUnrecognizedPattern ($"Expression call '{calledMethod.FullName}' inside '{callingMethodBody.Method.FullName}' was detected with 2nd argument which cannot be analyzed");
+									} else {
+										reflectionContext.RecordUnrecognizedPattern ($"Expression call '{calledMethod.FullName}' inside '{callingMethodBody.Method.FullName}' was detected with 2nd argument which cannot be analyzed");
+									}
+								}
+							}
+							break;
+
+						//
+						// static New (Type)
+						//
+						case "New" when calledMethod.DeclaringType.Name == "Expression"
+							&& calledMethod.DeclaringType.Namespace == "System.Linq.Expressions"
+							&& calledMethod.Parameters.Count == 1
+							&& calledMethod.Parameters[0].ParameterType.FullName == "System.Type":
+							{
+
+								reflectionContext.AnalyzingPattern();
+
+								foreach (var value in methodParams [0].UniqueValues ()) {
+									if (value is SystemTypeValue systemTypeValue)
+										MarkMethodsFromReflectionCall (ref reflectionContext, systemTypeValue.TypeRepresented, ".ctor", BindingFlags.Instance, parametersCount: 0);
```

Comment:
Technically this should include Public | NonPublic - since the runtime version will handle both cases.